packages/app/src/components-styled/vertical-bar-chart/logic/series.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  isDateSeries,
+  isDateSpanSeries,
+  TimestampedValue,
+} from '@corona-dashboard/common';
+import { useMemo } from 'react';
+import { pick } from 'lodash';
+import { isPresent } from 'ts-is-present';
+import { SeriesSingleValue } from '~/components-styled/time-series-chart/logic/series';
+
+export type SeriesConfig<T extends TimestampedValue> = BarSeriesDefinition<T>[];
+
+export type BarSeriesDefinition<T extends TimestampedValue> = {
+  type: 'bar';
+  metricProperty: keyof T;
+  label: string;
+  color: string;
+  secondaryColor?: string;
+};
+
+export function useSeriesList<T extends TimestampedValue>(
+  values: T[],
+  seriesConfig: SeriesConfig<T>
+) {
+  return useMemo(() => getSeriesList(values, seriesConfig), [
+    values,
+    seriesConfig,
+  ]);
+}
+
+export type SeriesList = SeriesSingleValue[][];
+
+export function getSeriesList<T extends TimestampedValue>(
+  values: T[],
+  seriesConfig: SeriesConfig<T>
+): SeriesList {
+  return seriesConfig.map((config) =>
+    getSeriesData(values, config.metricProperty)
+  );
+}
+
+/**
+ * This function is almost exactly like getSeriesData from TimeSeriesChart
+ * except it uses the date_end_unix as the date __date_unix in a date span
+ */
+export function getSeriesData<T extends TimestampedValue>(
+  values: T[],
+  metricProperty: keyof T
+): SeriesSingleValue[] {
+  if (values.length === 0) {
+    return [];
+  }
+
+  if (isDateSeries(values)) {
+    return values.map((x) => ({
+      __value: (x[metricProperty] ?? undefined) as number | undefined,
```

Comment:
```suggestion
      __value: x[metricProperty] as number | undefined,
```
Maybe not your code, but I think this does exactly the same

Reply:
It looks like this is necessary to fix a typescript error. It is also present on the TimeSeriesChart series.js file